src/main/java/net/finmath/time/ScheduleGenerator.java

```diff
@@ -625,6 +629,55 @@ public static ScheduleInterface createScheduleFromConventions(
 		// spotOffsetDays=0
 		return createScheduleFromConventions(referenceDate, 0, startOffsetString, maturityString, frequency, daycountConvention, shortPeriodConvention, dateRollConvention, businessdayCalendar, fixingOffsetDays, paymentOffsetDays);
 	}
+	
+	/**
+	 * Schedule generation with futureCodes. Futures are assumed to expire on the third wednesday in the respective month.
+	 * See other constructor for parameter description.
+	 * 
+	 * 
+	 * 
+	 * @param referenceDate The date which is used in the schedule to internally convert dates to doubles, i.e., the date where t=0.
+	 * @param futureCode Future code, e.g. DEC2017
+	 * @param startOffsetString The start date as an offset from the spotDate (build from tradeDate and spotOffsetDays) entered as a code like 1D, 1W, 1M, 2M, 3M, 1Y, etc.
+	 * @param maturityString The end date of the last period entered as a code like 1D, 1W, 1M, 2M, 3M, 1Y, etc.
+	 * @param frequency The frequency (as String).
+	 * @param daycountConvention The daycount convention (as String).
+	 * @param shortPeriodConvention If short period exists, have it first or last (as String).
+	 * @param dateRollConvention Adjustment to be applied to the all dates (as String).
+	 * @param businessdayCalendar Business day calendar (holiday calendar) to be used for date roll adjustment.
+	 * @param fixingOffsetDays Number of business days to be added to period start to get the fixing date.
+	 * @param paymentOffsetDays Number of business days to be added to period end to get the payment date.
+	 * @return The corresponding schedule
+	 */
+	public static ScheduleInterface createScheduleFromConventions(
+			LocalDate referenceDate,
+			String futureCode,
+			String startOffsetString,
+			String maturityString,
+			String frequency,
+			String daycountConvention,
+			String shortPeriodConvention,
+			String dateRollConvention,
+			BusinessdayCalendarInterface businessdayCalendar,
+			int	fixingOffsetDays,
+			int	paymentOffsetDays
+			)
+	{
+		int futureExpiryYears = 2000 + Integer.parseInt(futureCode.substring(futureCode.length()-2));
+		String futureExpiryMonthsString = futureCode.substring(0,futureCode.length()-2);
+		DateTimeFormatter formatter = new DateTimeFormatterBuilder().parseCaseInsensitive().appendPattern("dd/MMM/yyyy").toFormatter(Locale.ENGLISH);
+		String futureExpiryString = "01/" + futureExpiryMonthsString + "/" + futureExpiryYears;
+		LocalDate futureExpiryDate = LocalDate.parse(futureExpiryString, formatter);
```

Comment:
If the input string does not have the expected format, e.g.  'DEC2017', the parse function will raise an exception complaining about the format of "01/DEZ20/2017". This can be confusing for the user. Therefore I would wrap the conversion in a try block and raise a proper exception complaining about the format of maturityString.